src/main/io/gps.c

```diff
@@ -1337,14 +1340,51 @@ static void calculateNavInterval (void)
     gpsSol.navIntervalMs = constrain(navDeltaTimeMs, 50, 2500);
 }
 
+#if defined(USE_VIRTUAL_GPS)
+static void updateVirtualGPS(void)
+{
+    const uint32_t updateInterval = 100; // 100ms 10Hz update time interval
+    static uint32_t nextUpdateTime = 0;
+
+    if (cmp32(gpsData.now, nextUpdateTime) > 0) {
+        if (gpsData.state == GPS_STATE_INITIALIZED) {
+            gpsSetState(GPS_STATE_RECEIVING_DATA);
+        }
+
+        getVirtualGPS(&gpsSol);
+        gpsSol.time = gpsData.now;
+
+        gpsData.lastNavMessage = gpsData.now;
+        sensorsSet(SENSOR_GPS);
+
+        if (gpsSol.numSat > 3) {
+            gpsSetFixState(GPS_FIX);
+        } else {
+            gpsSetFixState(0);
+        }
```

Comment:
Perhaps ignore and leave for future improvement as current implementation use fixed satellites (12)
```suggestion
        gpsSetFixState(GPS_FIX);
```

Reply:
Thought about this too. 
But this simple imitation is working. 
While there are no connect to SITL, the setVirtualGPS does not run and numSat=0.  The configurator shows GPS NoFix.